proposals/split_kyverno_controllers.md

```diff
@@ -0,0 +1,120 @@
+# Meta
+[meta]: #meta
+- Name: Split / Decompose Kyverno Controllers
+- Start Date: November 9th, 2022
+- Update data (optional): November 9th, 2022
+- Author(s): @JimBugwadia, @eddycharlie
+
+# Table of Contents
+[table-of-contents]: #table-of-contents
+- [Meta](#meta)
+- [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+- [Implementation](#implementation)
+- [Migration (OPTIONAL)](#migration-optional)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Prior Art](#prior-art)
+- [Unresolved Questions](#unresolved-questions)
+- [CRD Changes (OPTIONAL)](#crd-changes-optional)
+
+# Overview
+[overview]: #overview
+
+Kyverno 1.8 runs as a single Deployment. This includes the webhook, reporting, and background controllers for generate and update requests. This design causes heavy backhround processing to impact the webhook performance. This proposal separates the webhook into its own Deployment.
+
+# Definitions
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that Kyverno users or other interested parties may not be familiar with.
+
+# Motivation
+[motivation]: #motivation
+
+- Eliminate, or greatly reduce, client-side throttling of the webhook.
+- Improve webhook performance
+- Allow independent sizing and scalability of webhook and background processing controllers
+
+# Proposal
+
+## Kyverno Controllers
+
+### Webhook Controller
+
+The webhook controller is responsible for:
+1. Processing AdmissionReview requests from the API Server
+2. Listening to policy and policyexception changes to maintain the policy cache
+
+### Background Controller
+
+The background controller is responsible for:
+1. Processing report changes
+2. Processing mutate and generate requests
+
+**NOTE 1**: its possible to further decompose the background controller, but seems like that may be of incremental value. Even with a single background controller, we can enable / disable modules or features.
```

Comment:
It makes sense to me to separate the reports system in a dedicated binary.
Especially in large clusters where a user could decide to shard the work between multiple instances.